### PR TITLE
Migrate xsvm from github.com/ava-labs/xsvm

### PR DESCRIPTION
**Pull request by marun**
## Why this should be merged

A simple VM is needed to enable subnet testing in e2e. [timestampvm](https://github.com/ava-labs/timestampvm) was originally suggested but xsvm ended up a better fit due to implementing [AWM](https://github.com/ava-labs/subnet-evm/blob/master/x/warp/README.md) which would also benefit from e2e testing. 

Having xsvm in-tree and running in CI will both enable better testing and ensure the VM remains compatible with the latest state of avalanchego.

## How this works

- Initial commit copies the code verbatim from https://github.com/ava-labs/xsvm (`Copy code from...`)
   - From avalanchego clone
   - `mdkir -p vms/example`
   - `git clone https://github.com/ava-labs/xsvm vms/example/xsvm`
   - `pushd vms/example/xsvm && git checkout -b cbeff25 && popd`
   -  `pushd vms/example/xsvm && rm -rf .g* && rm LICENSE* && rm go.* && rm -r scripts && popd`
- Subsequent commits:
  - Modify the code to build against latest avalanchego and in its new location (`Add build script...`)
  - Clean up the code to pass lint checks (`Update license header...` and `Ensure code conforms...`)
  - Update the README to reflect the move (`Update xsvm README...`)

## How this was tested

- Code builds
- e2e testing is intended to be added by #2043

## TODO
 - [x] Someone who knows what they are doing should probably rewrite the VM's `GetBlockIDAtHeight` and `VerifyHeightIndex` methods.

### CODE REVIEW

File: vms/example/xsvm/vm.go [hunk 1]
```diff
@@ -206,3 +206,44 @@ func (vm *VM) LastAccepted(context.Context) (ids.ID, error) {
...
+		if currentHeight < height {
+			break
+
+		}
+		currentID = currentBlock.ID()
+
+	}
+	return ids.ID{}, database.ErrNotFound
+
+}
```
Comment (StephenButtolph) [inline vms/example/xsvm/vm.go]: This isn't a realistic implementation. We should be indexing these heights as we accept blocks.

File: vms/example/xsvm/tx/export.go [hunk 2]
```diff
@@ -3,15 +3,14 @@
...
-)
+import "github.com/ava-labs/avalanchego/ids"
 
 var _ Unsigned = (*Export)(nil)
 
 type Export struct {
-	// ChainID provides cross chain replay protection
-	ChainID ids.ID `serialize:"true" json:"chainID"`
+	// NetworkID provides cross chain replay protection
+	NetworkID uint32 `serialize:"true" json:"networkID"`
```
Comment (StephenButtolph) [inline vms/example/xsvm/tx/export.go:12]: We don't need the networkID here.

File: scripts/build_xsvm.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+PLUGIN_FILENAME="v3m4wPxaHpvGr8qfMeyK6PRW3idZrPHmYcMTt7oXdK47yurVH"
+
+# Set the PATHS
+GOPATH="$(go env GOPATH)"
+if [[ $# -eq 1 ]]; then
+    BINARY_DIRECTORY=$1
+elif [[ $# -eq 0 ]]; then
+    BINARY_DIRECTORY="${GOPATH}/src/github.com/ava-labs/avalanchego/build/plugins/${PLUGIN_FILENAME}"
+else
```
Comment (StephenButtolph) [inline scripts/build_xsvm.sh:20]: Can we put this in the default plugin location? `~/.avalanchego/plugins`. Putting it into the avalanchego build directory is a remnant of the (very) old default plugin location.

**Review by StephenButtolph — APPROVED** [review]

**Review by hexfusion — APPROVED** [review]
So sweet thank you!

**Pull request merged**